flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/sink/KafkaRecordSerializationSchemaBuilder.java

```diff
@@ -369,5 +377,23 @@ public ProducerRecord<byte[], byte[]> serialize(
                     value,
                     headerProvider != null ? headerProvider.getHeaders(element) : null);
         }
+
+        @Override
+        public List<LineageDatasetFacet> getDatasetFacets() {
+            List<LineageDatasetFacet> facets = new ArrayList<>();
+            facets.add(new KafkaTopicListFacet(Arrays.asList(topicSelector.apply(null))));
```

Comment:
Is `topicSelector.apply(null)` guaranteed to work?
Is this even the right thing to do? TopicSelector could return different topics coming from different inputs.
I think we should instead check if TopicSelector is also a LineageProvider and ask it directly.
Our TopicSelectors should then implement it and we should add to the javadoc that LineageProvider is encouraged.

Reply:
Thanks for pointing this. It works only for the scenario `KafkaRecordSerializationSchema.builder().setTopic(DEFAULT_TOPIC)`. I've changed the implementation to more clear on this case. 

TopicSelector deserves more abstraction than being just `Function<? super IN, String>`, but I don't think this should be part of the scope of this PR.